src/cript/api/api.py

```diff
@@ -0,0 +1,234 @@
+import copy
+import os
+import warnings
+from typing import List, Literal, Union
+
+from cript.api._valid_search_modes import _VALID_SEARCH_MODES
+from cript.api.exceptions import CRIPTAPIAccessError, CRIPTConnectionError
+from cript.nodes.primary_nodes.primary_base_node import PrimaryBaseNode
+from cript.nodes.primary_nodes.project import Project
+from cript.nodes.supporting_nodes.group import Group
+from cript.nodes.supporting_nodes.user import User
+
+_global_cached_api = None
```

Comment:
It is not a singleton, since you can change the API access during the execution of your python program.
Instance might work, but I like cache better, since it just caches an otherwise existing object.